### DISCUSSION TITLE
Custom sources?

### DISCUSSION THREAD

**Discussion by mattwigham** [uploadcare/file-uploader#372] [blocks] [answered]
Is there a way to create [custom sources / tabs](https://uploadcare.com/docs/uploads/file-uploader/#configure-1) with blocks? Thanks.

**Comment by nd0ut** [comment, marked as answer]
Hey @mattwigham,

Yep, that's possible but undocumented yet. Here is an example of how to create custom "Unsplash" source: https://github.com/uploadcare/blocks-examples/tree/main/examples/composition/src/custom-source

Demo: https://uploadcare.github.io/blocks-examples/composition/src/custom-source/index.html

It's written using [symbiote.js](https://github.com/symbiotejs/symbiote.js) but you aren't restricted in use of any other rendering library inside symbiote.js component.
